index.js

```diff
@@ -43,6 +43,13 @@ load.path = function (dir) {
   var nearby = resolve(path.dirname(process.execPath))
   if (nearby) return nearby
 
+  // append platform, and prefix with scoped name matching package name if unscoped
+  var platformPackage = (packageName[0] === '@' ? '' : '@' + packageName + '/') + packageName + '-' + platform + '-' + arch
+  try {
+    var prebuildPackage = path.dirname(require('module').createRequire(path.join(dir, 'package.json')).resolve(platformPackage))
```

Comment:
Could we use `require.resolve()`? If that helps webpack, because `__non_webpack_require__.resolve()` should work in theory.

Reply:
I am pretty sure the reliance on package.json to determine this name is what will kill any chance of webpack working.
The `runtimeRequire(path.join(dir, 'package.json'))` call above will most likely either not find a file, or will find the one of the users project (not the library)

Is there a reason that this library tries to be clever in determining the name rather than requiring that as a parameter?

Reply:
This needs to be changed, or it crashes when package.json is not found
```
var platformPackage
  try {
    platformPackage = (packageName[0] === '@' ? '' : '@' + packageName + '/') + packageName + '-' + platform + '-' + arch
```

Reply:
> I am pretty sure the reliance on package.json to determine this name is what will kill any chance of webpack working. The `runtimeRequire(path.join(dir, 'package.json'))` call above will most likely either not find a file, or will find the one of the users project (not the library)
> 
> Is there a reason that this library tries to be clever in determining the name rather than requiring that as a parameter?

That's a separate discussion to be had (outside of this PR) as we'd need to drop some features (like multiple prebuilt binaries per platform, for different libuv versions, ABI's, etc) in order to remove discovery.

For now the approach is, any file to be found by `node-gyp-build` must be present on disk, outside of a webpack bundle.

Reply:
I agree it is a separate discussion. I dont think it would require dropping any features though. But Ill continue this discussion elsewhere.

I dont think this will make webpack support any worse, and will require the workarounds as suggested in #22

Reply:
I'm also fine with keeping this PR as-is @kriszyp; if you don't feel like exploring whether `require.resolve()` would be any better, then we'll tackle webpack later.

Reply:
I am actually not sure I follow which `require` you are talking about. Are you talking about node-gyp-build's `require`? Couldn't that be in an entirely different directory structure, and therefore have a different resolution path than the calling library's directory and resolution path? That's why I tried to recreate the require of the calling library to use it's resolve. Or is the idea that most of the time node-gyp-build will resolve the package same as the library's resolve? And it probably will most of the time, but I think if you have two different versions of the same package, that are using the same node-gyp-build version, it would likely resolve incorrectly.

I had also considered the possibility of taking `require` as an argument to node-gyp-build, so the library could provide its own `require` for us to use and resolve with. Maybe that would work best, I had just been avoiding any unnecessary API adjustments.

Reply:
> I am actually not sure I follow which `require` you are talking about. 

Instead of `require('module').createRequire(path.join(dir, 'package.json')).resolve(platformPackage)` you can do `require.resolve(platformPackage, { paths: [dir] })` ([docs](https://nodejs.org/api/modules.html#requireresolverequest-options)). But now that I see these side by side there's actually no difference for webpack; they do the same thing and have the same requirements. So nevermind :)